### DISCUSSION TITLE
Feature Request: Slumber Function

### DISCUSSION THREAD

**Discussion by Fenron-dev** [advplyr/audiobookshelf-app#24] [Ideas]
Hey, first off...thx for this awesome tool! 

One Feature I really miss in the app is a slumber function to stop playback after XX min. 
I mostly listen to audiobooks before sleeping, so this feature would save me from switching between different apps.

anyway, thx again and keep it up. :)

**Comment by advplyr** [comment]
Just released `v0.9.14-beta` with a sleep timer.
I included a 1 minute timer because it makes testing easier.

![image](https://user-images.githubusercontent.com/67830747/139555920-b0ea4234-43ff-4dfa-9491-bc1fad11bdf7.png)

Let me know if this is what you were looking for.

**Reply by p1xelll** [reply]
I am glad for this feature. It would be possible to add an "end of chapter" option.

**Reply by advplyr** [reply]
Just released `v0.9.15-beta` available on the play store with an end of chapter option.

I wasn't sure about how to display it so I just went with my first thought, let me know if you have feedback on that.

**Reply by p1xelll** [reply]
I would use time to end of chapter instead of EOC.

**Reply by advplyr** [reply]
Yeah I just didn't want it to be confused with the countdown timer. I guess it could be a negative timestamp. -04:30

**Reply by Fenron-dev** [reply]
Sorry for the late reply....thx again. Works exactly as I imagined it!

I have one more idea for an addition, if it's not too difficult. In the podcast app I use, it is possible to extend the snooze time again by the same number of minutes, once I shake the phone. Before the snooze ends, the volume goes down (in 5-8 sec) so I know when it's time to shake the phone again.

This way I don't have to open my eyes to restart playback.

**Reply by advplyr** [reply]
I still plan on adding this functionality. Also outlined in this issue #10

**Reply by Fenron-dev** [reply]
Ah, thx I missed that one. I'm glad you got that on your list already. :)

**Reply by advplyr** [reply]
Sleep timer was updated again in `v0.9.28-beta`, I posted about the update in issue #10
